### DISCUSSION TITLE
part-ii/2003-3/

### DISCUSSION THREAD

**Discussion by tripos-comments** [tripos-education/maths-tripos-questions#219] [Question comments]
Part II 2003 Graph Theory: A1.8 

https://questions.tripos.org/part-ii/2003-3/

**Comment by walrus1729** [comment]
<details>
<summary>Solution</summary>

(i) This is bookwork.

(ii) We use a similar method to the proof of Mantel's theorem. We double count

$$|\sum_{x \in X} \{(y,z) \in Y: xy, xz \in E(G)\}|$$

On the one hand, this is equal to $\sum_{x \in X} {d(x) \choose 2}$, and on the other hand, $y, z \in Y$ must have $|N(y) \cap N(z)| \leq 1$ (else we would get a $4$-cycle), so the sum is bounded above by ${n \choose 2}$ (summing over all pairs $(y,z)$). Then we apply Jensen's inequality to get that

$$n{e(G)/n \choose 2} \leq {n \choose 2}$$

and solving this as a quadratic in $e(G)$ gives the desired result.

A necessary condition for equality to be possible is for the upper bound to be an integer, which occurs for $n=1,3,7$ (for $n \leq 12$). In fact in all three of those cases, equality is possible; we have $K_2$, $C_6$ and the Heawood graph respectively.

</details>
